src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Presentation/Signature.cs

```diff
@@ -118,24 +192,22 @@ private void Initialize(bool setParameters)
             AddRange(_signatureHelpItem.DescriptionParts, parts, prettyPrintedParts);
             Append(_signatureHelpItem.DescriptionParts.GetFullText(), content, prettyPrintedContent);
 
-            if (_signatureHelpItem.Documentation.Count > 0)
+            // Note: This is where we realize the lazy Documentation. Be careful not to do so more than once.
+            var documentation = _signatureHelpItem.Documentation.ToArray();
```

Comment:
As the note says, I want to avoid enumerating more than once. If I used the IEnumerable, there would be one for the ".Count > 0" check (I would call the .Any() extension method), then there would be two more to call AddRange and Append on lines 202-203.

This is subtle and one of the things I was looking for feedback on. The types I changed are carefully selected to avoid early evaluation by switching to an iterator and threading IEnumerable<SymbolDisplayPart> through to this line. The downside is that, if you enumerate more than once, you incur a 2nd parsing cost.

Now, this could probably be fixed by using a custom iterator type. One that is lazy (doesn't do any work until you call GetEnumerator), but fast for subsequent calls.